pkg/monitortests/machines/watchmachines/monitortest.go

```diff
@@ -0,0 +1,55 @@
+package watchmachines
+
+import (
+	"context"
+	"time"
+
+	machineClient "github.com/openshift/client-go/machine/clientset/versioned"
+	"github.com/openshift/origin/pkg/monitortestframework"
+
+	"github.com/openshift/origin/pkg/monitor/monitorapi"
+	"github.com/openshift/origin/pkg/test/ginkgo/junitapi"
+	"k8s.io/client-go/rest"
+)
+
+type machineWatcher struct {
+}
+
+func NewMachineWatcher() monitortestframework.MonitorTest {
+	return &machineWatcher{}
+}
+
+func (w *machineWatcher) StartCollection(ctx context.Context, adminRESTConfig *rest.Config, recorder monitorapi.RecorderWriter) error {
+	machineClient, err := machineClient.NewForConfig(adminRESTConfig)
+	if err != nil {
+		return err
+	}
+
+	startMachineMonitoring(ctx, recorder, machineClient)
+
+	return nil
+}
+
+func (w *machineWatcher) CollectData(ctx context.Context, storageDir string, beginning, end time.Time) (monitorapi.Intervals, []*junitapi.JUnitTestCase, error) {
+	// because we are sharing a recorder that we're streaming into, we don't need to have a separate data collection step.
+	return nil, nil, nil
+}
+
+func (*machineWatcher) ConstructComputedIntervals(ctx context.Context, startingIntervals monitorapi.Intervals, recordedResources monitorapi.ResourcesMap, beginning, end time.Time) (monitorapi.Intervals, error) {
```

Comment:
this function can take all the edges/instants from the collection and it can produce the intervals for individual machines.

Reply:
I don't really understand what this means. Can you please elaborate more on this?